languages/tree-sitter-stack-graphs-java/src/stack-graphs.tsg

```diff
@@ -141,6 +146,35 @@ attribute node_symbol = node            => symbol = (source-text node), source_n
     edge @class.type -> ref
 }
 
+
+(class_declaration (type_parameters)) @class {
+  node @class.type_parameters
+  edge @class.lexical_scope -> @class.type_parameters
+}
+
+(class_declaration (type_parameters (type_parameter) @param)) @class {
+  edge @class.type_parameters -> @param.def
+  edge @param.lexical_scope -> @class.lexical_scope
+}
+
+(type_parameter (type_identifier) @id) @this {
+  node @this.def
+  node @this.lexical_scope
+
+  edge @id.lexical_scope -> @this.lexical_scope
+
+  attr (@this.def) node_definition = @id
+}
+
+(spread_parameter) @spread_param {
+  node @spread_param.lexical_scope
+  node @spread_param.def
+}
+
+(class_declaration interfaces: (super_interfaces (type_list (_) @type))) @this {
+  edge @type.lexical_scope -> @this.lexical_scope
+}
```

Comment:
I wouldn't mind seeing tsg supply us with `@this` automatically, but in the meantime I've taken to using it in many of these rules for consistency's sake.